dbus-wrapper/dbus_client.c

```diff
@@ -0,0 +1,328 @@
+/*
+ *   Software Updater - D-Bus client for the daemon controlling
+ *                      Clear Linux Software Update Client.
+ *
+ *      Copyright © 2016 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *   Contact: Dmitry Rozhkov <dmitry.rozhkov@intel.com>
+ */
+
+#include <errno.h>
+
+#include <systemd/sd-bus.h>
+#include <systemd/sd-event.h>
+
+#include "dbus_client.h"
+#include "option.h"
+#include "log.h"
+
+typedef struct _command_ctx {
+	sd_bus *bus;
+	const char *method;
+	struct list *opts;
+	dbus_cmd_argv_type argv_type;
+	char **argv;
+} command_ctx_t;
+
+static int on_child_output_received(sd_bus_message *message, void *userdata, sd_bus_error *error)
+{
+	const char *output;
+	int r;
+
+	r  = sd_bus_message_read(message, "s", &output);
+	if (r < 0) {
+		ERR("Can't read client's output: %s", strerror(-r));
```

Comment:
Yet I don't like using pure printf()s for error reporting, because in a bigger code base grepping for the error message may result in multiple findings. It's easier to modify the macro to include file name and line number then the failure point becomes obvious.

I'll remove file and line number from the macro.